static/src/javascripts/projects/commercial/commercial-metrics.ts

```diff
@@ -11,20 +12,28 @@ let logged = false;
 
 const isDev = Boolean(config.get('page.isDev', false));
 
-const init = (): void => {
-	if (!window.guardian.ophan) return;
-	if (!config.get('switches.commercialMetrics', false)) return;
+const init = (): Promise<void> => {
+	if (!window.guardian.ophan) return Promise.resolve();
+	if (!config.get('switches.commercialMetrics', false))
+		return Promise.resolve();
 
 	const userIsInSamplingGroup = Math.random() <= 0.01;
+	const shouldForceMetrics = getSynchronousTestsToRun().some((test) =>
+		// The convention is that if a test ID starts with “Commercial”,
+		// we track all the commercial metrics for it
+		test.id.startsWith('Commercial'),
```

Comment:
I can see this behavior cropping up as an unintended side effect in the future if someone who wasn't aware of it were to create an AB test with a name starting with `Commercial` but for which they didn't want to force commercial metrics.
If we can't have an explicit property like `test.properties.forceMetrics` then maybe for now we should consider only forcing metrics if test.id == 'CommercialPartner'?

Reply:
Ok, that’s a fair point. A clean solution would be: 

```ts

import {commercialPartner, anotherPartner, superPerformanceTest} from './ab-tests.ts';

const testsWithCommercialMetrics = [
    commercialPartner,
    anotherPartner,
    superPerformanceTest
];

const shouldForceMetrics = getSynchronousTestsToRun().some((test) =>
    testsWithCommercialMetrics.map(t => t.id).includes(test.id)
);
``` 

This means that it's up to our commercial metrics definition to keep up with which tests should be included, and nothing it entered as individual strings so changing the name of a test or a capitalisation won’t result in a severe issue.